src/js/parse-content.js

```diff
@@ -29,11 +29,12 @@ module.exports = function parseContent(options) {
   /**
    * Get important properties from a heading element and store in a plain object.
    * @param {HTMLElement} heading
+   * @param {String} tocHrefAttribute
    * @return {Object}
    */
-  function getHeadingObject(heading) {
+  function getHeadingObject(heading, tocHrefAttribute) {
     var obj = {
-      id: heading.id,
```

Comment:
Not setting the id will break the default hash handling of the browser.

Your way will work for users who click in the Toc to jump to a section. And it should work so the proper section is highlighted as you scroll. But what about back / forward buttons within a page? I think this breaks that functionality when the tocHrefAttribute is used?